plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/inline/InlineChatActionPromoter.kt

```diff
@@ -0,0 +1,35 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cwc.inline
+
+import com.intellij.openapi.actionSystem.ActionPromoter
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.keymap.KeymapUtil
+import com.intellij.openapi.util.SystemInfo
+
+class InlineChatActionPromoter : ActionPromoter {
+    // temporary until we find a better key binding
+    override fun promote(actions: MutableList<out AnAction>, context: DataContext): MutableList<AnAction> {
+        val results = actions.toMutableList()
+        if (context.getData(CommonDataKeys.EDITOR) == null || context.getData(CommonDataKeys.PROJECT) == null) return results
+        val shortCut = KeymapUtil.getShortcutText("aws.toolkit.jetbrains.core.services.cwc.inline.openChat")
+        // only promote for the default key bindings
+        if (SystemInfo.isMac && shortCut != "⌘I") return results
+        if (!SystemInfo.isMac && shortCut != "Ctrl+I") return results
+
+        results.sortWith { a, b ->
+            when {
+                isOpenChatInputAction(a) -> -1
+                isOpenChatInputAction(b) -> 1
+                else -> 0
+            }
+        }
+        return results
```

Comment:
```suggestion
        return actions.sortedBy { a, b ->
            when {
                isOpenChatInputAction(a) -> -1
                isOpenChatInputAction(b) -> 1
                else -> 0
            }
        }
```

Reply:
should it be
```
                isOpenChatInputAction(a) -> -1
                else -> 0
```
?

Reply:
it might be redundant but feel like it's safer

Reply:
went with the original since it's a mutable list